src/status_im/ui/screens/desktop/main/chat/views.cljs

```diff
@@ -189,30 +194,41 @@
      [chat-text-input]]))
 
 (views/defview chat-profile []
-  (views/letsubs [{:keys [pending? whisper-identity public-key] :as contact} [:get-current-chat-contact]]
-                 [react/view {:style styles/chat-profile-body}
-                  [profile.views/profile-badge contact]
-                  ;; for private chat, public key will be chat-id
-                  [react/view
-                   (if pending?
-                     [react/touchable-highlight {:on-press #(re-frame/dispatch [:add-contact whisper-identity])}
+  (views/letsubs [current-chat-contact   [:get-current-chat-contact]
+                  already-stored-contact [:get-current-contact]
+                  contact-identity       [:get-current-contact-identity]]
+                 ;; handle three possible cases
+                 ;; 1. coming from a private chat with all contact information readily available
+                 ;; 2. coming from public chat with already saved user info in contacts
+                 ;; 3. coming from a public chat with nothing but a whisper identity
+                 (let [{:keys [pending? whisper-identity] :as contact} (or current-chat-contact
+                                                                           already-stored-contact
+                                                                           (utils.contacts/whisper-id->new-contact contact-identity))]
+                   [react/view
+                    [react/view {:style styles/desktop-profile-back-button}
+                     toolbar.view/default-nav-back]
+                    [profile.views/profile-badge contact]
+                    [react/view {:style styles/chat-profile-body}
+                     ;; pending? will come back as nil for public chat so explicitly check for false here
+                     (if (false? pending?)
```

Comment:
`if-not` ?

Reply:
Thanks for bringing this into the light. I *think* a check for an explicit false is appropriate here as the desired behavior is to have `nil` handled the same as `true`. (i.e. both indicative of pending? not having been manually updated to false.) 

I've added more comments inline to address the intention here. Let me know if something still seems off.